src/lib/middleware/framework-services/fastify.ts

```diff
@@ -0,0 +1,45 @@
+import { FrameworkService } from "../interfaces";
+
+export default class FastifyService implements FrameworkService {
+
+    getReqHeaderField(req: any, fieldName: string): string | undefined {
+        return req.headers[fieldName]
+    }
+
+    getReqField(req: any, fieldName: string): any {
+        let value: string | number | boolean | undefined = undefined;
+        switch (fieldName) {
+            case "protocol":
+                value = "HTTP" + (req.raw.httpVersion == null ? "" : "/" + req.raw.httpVersion);
+                break;
+            case "remote_host":
+                value = req.raw.connection?.remoteAddress;
+                break;
+            case "remote_port":
+                value = req.raw.connection?.remotePort?.toString();
+                break;
+            case "remote_user":
+                if (req.raw.user && req.raw.user.id) {
+                    value = req.raw.user.id;
+                }
+                break;
+            default:
+                value = req.raw[fieldName]
+                break;
```

Comment:
Did you test all of these to work as expected?

Reply:
yes, I did. Fastify wraps the Node.js HTTP request object and stores it in the `.raw` object.